src/Options/Options.hpp

```diff
@@ -145,3 +147,25 @@ struct create_from_yaml {
   template <typename Metavariables>
   static T create(const Option& options);
 };
+
+namespace Options_detail {
+template <typename T, typename = cpp17::void_t<>>
+struct name_helper {
+  static std::string name() noexcept { return pretty_type::short_name<T>(); }
+};
+
+template <typename T>
+struct name_helper<T, cpp17::void_t<decltype(T::name())>> {
+  static std::string name() noexcept { return T::name(); }
+};
+template <typename T>
+std::string name() noexcept {
+  return Options_detail::name_helper<T>::name();
+}
+}  // namespace Options_detail
+
+// The name in the YAML file for a struct.
+template <typename T>
+std::string option_name() noexcept {
+  return Options_detail::name<T>();
```

Comment:
Use `name_helper` directly and get rid of the `name` detail function.  (Everything can just call `option_name`.)